invenio_previewer/extensions/mistune.py

```diff
@@ -22,8 +22,12 @@ def render(file):
     """Render HTML from Markdown file content."""
     with file.open() as fp:
         encoding = detect_encoding(fp, default="utf-8")
-        result = mistune.markdown(fp.read().decode(encoding))
-        return result
+        try:
+            result = mistune.markdown(fp.read().decode(encoding))
+            return result
+        except Exception as e:
```

Comment:
nit: I would move the error handling to the caller. Then the caller can decide what to do with it (i.e. log an error, render a default)